plugins/inputs/syslog/syslog.go

```diff
@@ -156,21 +156,45 @@ func (s *Syslog) Stop() {
 func (s *Syslog) createStreamDataHandler(acc telegraf.Accumulator) socket.CallbackConnection {
 	// Create parser options
 	var opts []syslog.ParserOption
-	if s.BestEffort {
-		opts = append(opts, syslog.WithBestEffort())
+	var machineOpts []syslog.MachineOption
+	switch s.SyslogStandard {
+	case "RFC3164":
+		if s.BestEffort {
+			machineOpts = append(machineOpts, rfc3164.WithBestEffort())
+		}
+		machineOpts = append(machineOpts, rfc3164.WithYear(rfc3164.CurrentYear{}))
+
+	case "RFC5424":
+		if s.BestEffort {
+			machineOpts = append(machineOpts, rfc3164.WithBestEffort())
+		}
```

Comment:
Copy-paste error in RFC5424 section.
```suggestion
	case "RFC5424":
		if s.BestEffort {
			machineOpts = append(machineOpts, rfc5424.WithBestEffort())
		}
```